include/nbl/builtin/hlsl/bxdf/transmission/smooth_dielectric.hlsl

```diff
@@ -115,29 +97,18 @@ struct SSmoothThinDielectric
 
     NBL_CONSTEXPR_STATIC_INLINE BxDFClampMode _clamp = BxDFClampMode::BCM_ABS;
 
-    struct SCreationParams
-    {
-        spectral_type eta2;
-        spectral_type luminosityContributionHint;
-    };
-    using creation_type = SCreationParams;
-
-    static this_t create(NBL_CONST_REF_ARG(spectral_type) eta2, NBL_CONST_REF_ARG(spectral_type) luminosityContributionHint)
+    static this_t create(NBL_CONST_REF_ARG(fresnel::Dielectric<spectral_type>) f, NBL_CONST_REF_ARG(spectral_type) luminosityContributionHint)
     {
         this_t retval;
-        retval.eta2 = eta2;
+        retval.fresnel = f;
         retval.luminosityContributionHint = luminosityContributionHint;
         return retval;
     }
```

Comment:
why have a `create` which fills all the members, only have a `create` if you do at least one thing of convenience